UnitTests/BraintreePaymentFlowTests/BTLocalPayment_Tests.swift

```diff
@@ -222,7 +223,77 @@ class BTLocalPayment_UnitTests: XCTestCase {
+        mockAPIClient.cannedResponseBody = BTJSON(
+            value: [
+                "paymentResource": [
+                    "redirectUrl": "https://www.somebankurl.com",
+                    "paymentToken": "123aaa-123-543-777",
+                ]
+            ]
+        )
+
+        client.startPaymentFlow(localPaymentRequest) { _, _ in }
+
+        let analyticMessage = "local-payment:" + BTPaymentFlowAnalytics.browserPresentationSucceeded
+        XCTAssertTrue(mockAPIClient.postedAnalyticsEvents.contains(analyticMessage))
+    }
+    
+    func testStartPayment_alert_cancel_sendsAnalyticEvent() {
+        mockAPIClient.cannedConfigurationResponseBody = BTJSON(value: [ "paypalEnabled": true ])
+        let mockWebAuthenticationSession = MockWebAuthenticationSession()
+        mockWebAuthenticationSession.cannedSessionDidDisplay = false
+        mockWebAuthenticationSession.cannedResponseURL =  nil
+        
+        mockWebAuthenticationSession.cannedErrorResponse = ASWebAuthenticationSessionError(
+            _bridgedNSError: NSError(
+                domain: ASWebAuthenticationSessionError.errorDomain,
+                code: ASWebAuthenticationSessionError.canceledLogin.rawValue,
+                userInfo: ["Description": "Mock cancellation error description."]
+            )
+        )
+        
+        let client = BTPaymentFlowClient(apiClient: mockAPIClient)
+        client.returnedToAppAfterPermissionAlert = false
+        client.webAuthenticationSession = mockWebAuthenticationSession
+        mockAPIClient.cannedResponseBody = BTJSON(
+            value: [
+                "paymentResource": [
+                    "redirectUrl": "https://www.somebankurl.com",
+                    "paymentToken": "123aaa-123-543-777",
+                ]
+            ]
+        )
+
+        client.startPaymentFlow(localPaymentRequest) { _, _ in }
+
+        let analyticMessage = "local-payment:" + BTPaymentFlowAnalytics.browserLoginAlertCanceled
+        XCTAssertTrue(mockAPIClient.postedAnalyticsEvents.contains(analyticMessage))
+    }
+    
+    func testStartPayment_browser_cancel_sendsAnalyticEvent() {
+        mockAPIClient.cannedConfigurationResponseBody = BTJSON(value: [ "paypalEnabled": true ])
+        let mockWebAuthenticationSession = MockWebAuthenticationSession()
+        mockWebAuthenticationSession.cannedSessionDidDisplay = true
+        mockWebAuthenticationSession.cannedResponseURL =  nil
+        
+        mockWebAuthenticationSession.cannedErrorResponse = ASWebAuthenticationSessionError(
+            _bridgedNSError: NSError(
+                domain: ASWebAuthenticationSessionError.errorDomain,
+                code: ASWebAuthenticationSessionError.canceledLogin.rawValue,
+                userInfo: ["Description": "Mock cancellation error description."]
+            )
+        )
+        
         let client = BTPaymentFlowClient(apiClient: mockAPIClient)
+        client.returnedToAppAfterPermissionAlert = true
```

Comment:
I have to figure out a way that this is toggled during the ASWebSession due to system notification.

Reply:
This test was just to toss an idea but this test will always pass since I am resetting returnedToAppAfterPermit to false before the webSession launches in the code.